components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java

```diff
@@ -2317,11 +2317,14 @@ public ServiceProvider getApplicationByResourceId(String resourceId, String tena
         ApplicationDAO appDAO = ApplicationMgtSystemConfig.getInstance().getApplicationDAO();
         ServiceProvider application = appDAO.getApplicationByResourceId(resourceId, tenantDomain);
         if (application == null) {
-            if (log.isDebugEnabled()) {
-                log.debug("Cannot find an application for resourceId: " + resourceId + " in tenantDomain: "
-                        + tenantDomain);
+            application = new FileBasedApplicationDAO().getApplicationByResourceId(resourceId, tenantDomain);
+            if (application == null) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Cannot find an application for resourceId: " + resourceId + " in tenantDomain: "
```

Comment:
```suggestion
                    log.debug("Cannot find a file-based application with resource ID: " + resourceId + " in tenant domain: "
```

Reply:
We are not only checking in the file based application but also in cache and in database as well.